examples/Extensions/AddAffiliateLocationExtensions.cs

```diff
@@ -0,0 +1,439 @@
+        /// <param name="chainId">The retail chain ID.</param>
+        /// <returns>Resource name of the newly created Affiliate Location Extension feed.</returns>
+        private static string CreateALEFeed(GoogleAdsClient client, long customerId, long chainId)
+        {
+            // Optional: Delete all existing location extension feeds. This is an optional step,
+            // and is required for this code example to run correctly more than once.
+            // 1. Google Ads only allows one location extension feed per email address.
+            // 2. A Google Ads account cannot have a location extension feed and an affiliate
+            // location extension feed at the same time.
+            DeleteLocationExtensionFeeds(client, customerId);
+
+            // Get the FeedServiceClient.
+            FeedServiceClient feedService = client.GetService(Services.V3.FeedService);
+
+            // Creates a feed that will sync to retail addresses for a given retail chain ID.
+            // Do not add FeedAttributes to this object as Google Ads will add
+            // them automatically because this will be a system generated feed.
+            Feed aleFeed = new Feed()
+            {
+                Name = "Affiliate Location Extension feed #" + ExampleUtilities.GetRandomString(),
+
+                AffiliateLocationFeedData = new AffiliateLocationFeedData()
+                {
+                    ChainIds = { chainId },
+                    RelationshipType = AffiliateLocationFeedRelationshipType.GeneralRetailer
+                },
+                // Since this feed's feed items will be managed by Google,
+                // you must set its origin to GOOGLE.
+                Origin = FeedOrigin.Google
+            };
+
+            FeedOperation operation = new FeedOperation()
+            {
+                Create = aleFeed
+            };
+
+            // Adds the feed.
+            MutateFeedsResponse response =
+                feedService.MutateFeeds(customerId.ToString(), new[] { operation });
+
+            // Displays the results.
+            string aleFeedResourceName = response.Results[0].ResourceName;
+            Console.WriteLine($"ALE feed created with resource name: {aleFeedResourceName}.");
+            return aleFeedResourceName;
+        }
+
+        /// <summary>
+        /// Deletes the old location extension feeds.
+        /// </summary>
+        /// <param name="client">The Google Ads client.</param>
+        /// <param name="customerId">The customer ID for which the call is made.</param>
+        private static void DeleteLocationExtensionFeeds(GoogleAdsClient client, long customerId)
+        {
+            // To delete a location extension feed, you need to
+            // 1. Delete the CustomerFeed so that the location extensions from the feed stop
+            // serving.
+            // 2. Delete the feed so that Google Ads will no longer sync from the GMB account.
```

Comment:
I do not thing ALEs are coming from GMB account.

Reply:
This comment is valid here, because you have to delete both GMB and ALE feeds for this to work. An account can only have one of the two, and one ENABLED of a given type.

Reply:
SG